src/hotspot/cpu/x86/gc/z/zBarrierSetAssembler_x86.cpp

```diff
@@ -78,15 +78,43 @@ class ZRuntimeCallSpill {
 
   void save() {
     MacroAssembler* masm = _masm;
-    __ push(rax);
-    __ push(rcx);
-    __ push(rdx);
-    __ push(rdi);
-    __ push(rsi);
-    __ push(r8);
-    __ push(r9);
-    __ push(r10);
-    __ push(r11);
+    if (VM_Version::supports_apx_f()) {
+      __ push(rax);
```

Comment:
if _result is not equal to rax this also could be pushp rax here and popp rax in restore().